xla/service/gpu/runtime/nccl_collective_permute_thunk.cc

```diff
@@ -217,6 +299,14 @@ absl::Status RunCollectivePermute(
                                   device_string);
     TF_RETURN_IF_ERROR(stream.MemZero(&dest_addr, dest_addr.size()));
   }
+  if (use_memcpy && target_id) {
+    // Need to block until the receive pointer buffer has been updated.
+    TF_RETURN_IF_ERROR(stream.BlockHostUntilDone());
```

Comment:
One problem with this is that it will very likely will lead to deadlocks in presence of pending collective operations on different ranks, and it's incompatible with cuda graphs. Is it ok to give up cuda graphs for send/recv?

Reply:
Changing to AsyncValue got rid of this, now it blocks on the async event :). 
As for the cuda graph, most of the cases we dont get many CP kernels in the graph, it should be okay to exclude them from cuda graph. cuda graph might get rid of some launch overhead but compared to 3x speedup using p2p, it's a good tradeoff.